azuread/data_users.go

```diff
@@ -62,54 +127,83 @@ func dataSourceUsersRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ArmClient).usersClient
 	ctx := meta.(*ArmClient).StopContext
 
-	var users []graphrbac.User
+	var users []*graphrbac.User
 	expectedCount := 0
 
+	ignoreMissing := d.Get("ignore_missing").(bool)
 	if upns, ok := d.Get("user_principal_names").([]interface{}); ok && len(upns) > 0 {
 		expectedCount = len(upns)
 		for _, v := range upns {
-			resp, err := client.Get(ctx, v.(string))
+			u, err := client.Get(ctx, v.(string))
 			if err != nil {
-				return fmt.Errorf("Error making Read request on AzureAD User with ID %q: %+v", v.(string), err)
+				if !ignoreMissing && ar.ResponseWasNotFound(u.Response) {
```

Comment:
@patrickmarabeas Thanks for making those changes. I realize I didn't think out the logic here - my intention was to have the loop break if we're ignoring missing users and the error was a 404. Sorry about that, how about this instead?

```hcl
if ignoreMissing && ar.ResponseWasNotFound(u.Response) {
	break
}
return fmt.Errorf("Error making Read request on AzureAD User with ID %q: %+v", v.(string), err)
```

Reply:
Updated these blocks.